packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -782,7 +792,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   TextInputConnection _textInputConnection;
   TextSelectionOverlay _selectionOverlay;
 
-  final ScrollController _scrollController = ScrollController();
+  ScrollController get _scrollController => widget.scrollController ?? ScrollController();
```

Comment:
This would create new ScrollController on each get invocation, wouldn't it?
May be it should be:
```
final ScrollController _scrollController = widget.scrollController ?? ScrollController();
```

Reply:
Yes, you're right. 
The problem is that it is impossible to access widget from an initializer, I'll try a different approach may be working on initState.

Reply:
Or maybe something like this?
```
final ScrollController _defaultScrollController = ScrollController();
ScrollController get _scrollController => widget.scrollController ?? _defaultScrollController;
```

Reply:
This will needlessly allocate ScrollController when it is defined by widget.
Just remove `final` and initialize field in initState() as all `_cursorBlinkOpacityController` controller does.

Reply:
Updated with your suggestions